massa-models/src/block_header.rs

```diff
@@ -451,3 +467,23 @@ impl std::fmt::Display for BlockHeader {
         Ok(())
     }
 }
+
+/// A denunciation data for block header
+pub struct BlockHeaderDenunciationData {
+    slot: Slot,
+}
+
+impl BlockHeaderDenunciationData {
+    /// Create a new DenunciationData for block hedader
+    pub fn new(slot: Slot) -> Self {
+        Self { slot }
+    }
+
+    /// Get byte array
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut buf = Vec::new();
+        buf.extend(self.slot.period.to_le_bytes());
+        buf.push(self.slot.thread);
+        buf
```

Comment:
you can use directly `self.slot.to_bytes_key()`

Reply:
Done